src/sparseml/pytorch/sparsification/quantization/modifier_quantization.py

```diff
@@ -39,9 +58,180 @@
 def _select_quantization_modifier(state: Dict[str, Any]) -> Type:
     # if kwargs for the legacy quantization modifier are provided,
     # route YAML loading to that class
+    import pdb
+
+    pdb.set_trace()
     return LegacyQuantizationModifier if "submodules" in state else QuantizationModifier
 
 
 @PyTorchModifierYAML(swap_class_by_state_fn=_select_quantization_modifier)
 class QuantizationModifier(ScheduledModifier):
-    pass
+    """
+    Enables quantization aware training (QAT) for a given module or its submodules
+    After the start epoch, the specified module(s)' forward pass will emulate
+    quantized execution and the modifier will be enabled until training is completed.
+
+    | Sample yaml:
+    |   !QuantizationModifier
+    |       start_epoch: 0.0
+    |       default_scheme:
+    |           input_activations:
+    |               num_bits: 8
+    |               symmetric: False
+    |           weights:
+    |               num_bits: 8
+    |               symmetric: True
+
+    :param start_epoch: The epoch to start the modifier at
+    :param default_scheme: Default QuantizationScheme to use when enabling quantization
+        in a module. May also be a dictionary to be loaded into the QuantizationScheme
+        class. If None, the default scheme (`QuantizationScheme()`) will be used.
+        Default is None
+    :param end_epoch: Disabled, setting to anything other than -1 will raise an
+        exception. For compatibility with YAML serialization only.
+    """
+
+    def __init__(
+        self,
+        start_epoch: float = -1.0,
+        default_scheme: Union[QuantizationScheme, DictQuantizationScheme, None] = None,
+        end_epoch: float = -1,
```

Comment:
to enforce that the default is a float
```suggestion
        end_epoch: float = -1.0,
```